### ISSUE TITLE
bug: When using traffic-split ,  the key of chash is missing in the upstream of traffic-split

### ISSUE THREAD

**Issue by nanamikon** [apache/apisix#3858]
### Environment

* apisix version (2.3):
* OS (ubuntu 16):
* OpenResty / Nginx version (1.19.3):
* etcd version, if have (3.14):
* apisix-dashboard version, if have:

### Minimal test code / Steps to reproduce the issue

1.use traffic-split , and set the upstream as follow
```
curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "uri": "/index.html",
    "plugins": {
        "traffic-split": {
            "rules": [
                {
                    "match": [
                        {
                            "vars": [
                                ["http_release","==","new_release"]
                            ]
                        }
                    ],
                    "weighted_upstreams": [
                        {
                            "upstream": {
                                "type": "chash",
                                "key": "remote_addr",
                                "name": "upstream_A",
                                "nodes": {
                                    "127.0.0.1:1981":1
                                    "127.0.0.2:1981":1
                                }
                            }
                        }
                    ]
                }
            ]
        }
    },
    "upstream": {
            "type": "roundrobin",
            "nodes": {
                "127.0.0.1:1980": 1
            }
    }
}'
```

2.
curl http://127.0.0.1:9080/index.html -H "release:new_release" -i
```
<html>
<head><title>500 Internal Server Error</title></head>
<body>
<center><h1>500 Internal Server Error</h1></center>
<hr><center>openresty</center>
</body>
</html>
```

### What's the actual result? (including assertion message & call stack if applicable)

Error can be found in error.log
```
2021/03/18 15:36:04 [error] 72662#72662: *683 failed to run balancer_by_lua*: /chash.lua:36: invalid argument, expect string value
stack traceback:
        [C]: in function 'error'
        /data/app/huya-nginx-proxy/apisix/core/ctx.lua:139: in function '__index'
        /data/app/huya-nginx-proxy/apisix/balancer/chash.lua:36: in function 'fetch_chash_hash_key'
        /data/app/huya-nginx-proxy/apisix/balancer/chash.lua:81: in function 'get'
        /data/app/huya-nginx-proxy/apisix/balancer.lua:172: in function 'pick_server'
        /data/app/huya-nginx-proxy/apisix/balancer.lua:197: in function 'load_balancer'
        /data/app/huya-nginx-proxy/apisix/init.lua:718: in function 'http_balancer_phase'
```

### What's the expected result?
No error

**Comment by Firstsawyou** [comment]
The upstream configuration of the traffic-split plugin currently only supports the `roundrobin` method, and the chash method is not yet supported.
You can first use the upstream_id in the plugin to support the chash method.

<img width="1067" alt="截屏2021-03-18 下午4 14 57" src="https://user-images.githubusercontent.com/52862365/111593953-755a5380-8805-11eb-87b0-5bc80a5b5a15.png">

**Comment by nanamikon** [comment]
Thinks for reply,  I think my version 2.3 is not support upstream_id yet.   Maybe I should wait the next apisix version，  will merge this bug fix #3824

**Comment by Firstsawyou** [comment]
> Thinks for reply, I think my version 2.3 is not support upstream_id yet. Maybe I should wait for the next apisix version， will merge this bug fix #3824

Yes, 2.3 does not support upstream_id yet. It is best to wait for the next apisix version. At the same time, we will support the chash method as soon as possible.
